storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTask.java

```diff
@@ -90,12 +91,13 @@
     private Set<TopicIdPartition> assignedTopicPartitions = Collections.emptySet();
 
     // Map of remote log metadata topic partition to consumed offsets.
-    private final Map<Integer, Long> partitionToConsumedOffsets = new ConcurrentHashMap<>();
+    private final ConcurrentMap<Integer, Long> partitionToConsumedOffsets = new ConcurrentHashMap<>();
     private Map<Integer, Long> committedPartitionToConsumedOffsets = Collections.emptyMap();
 
     private final long committedOffsetSyncIntervalMs;
     private CommittedOffsetsFile committedOffsetsFile;
-    private long lastSyncedTimeMs;
+    private volatile long lastSyncedTimeMs;
+    private final Object syncCommittedDataLock = new Object();
```

Comment:
Would there be multiple threads calling this ConsumerTask?
It's not obvious why this lock was added.

Reply:
That is a good point. My suspicion is that there is some method that is exposed through the ConsumerManager that would modify one of the variables read or written inside the `synchronized` block. I tried finding it, but was unable to. Will look further or ask the Uber team.